mypyc/irbuild/specialize.py

```diff
@@ -75,6 +78,11 @@ def translate_len(
             # though we still need to evaluate it.
             builder.accept(expr.args[0])
             return builder.add(LoadInt(len(expr_rtype.types)))
+        elif is_list_rprimitive(expr_rtype):
+            obj = builder.accept(expr.args[0])
+            elem_address = builder.add(GetElementPtr(obj, PyVarObject, 'ob_size'))
+            size_value = builder.add(LoadMem(c_pyssize_t_rprimitive, elem_address))
+            return builder.call_c(size_t_to_short_int, [size_value], -1)
```

Comment:
We don't want to call a function here, since the reason we are doing this the hard way here instead of a primitive is to avoid a function call. Add a low-level IR builder helper for doing this. You can use `BinaryIntOp` to shift left by 1.